cmd/artist.go

```diff
@@ -50,20 +51,26 @@ func getArtists(username string, period Period, count int, imageSize string) ([]
 	r := *result
 
 	artists := make([]*Artist, len(r.TopArtists.Artists))
+	var wg sync.WaitGroup
+	wg.Add(len(artists))
 	for i, artist := range r.TopArtists.Artists {
 		newArtist := &Artist{
 			Name:      artist.Name,
 			Playcount: artist.Playcount,
 		}
 
-		for _, image := range artist.Images {
-			if image.Size == imageSize {
-				newArtist.ImageUrl = image.Link
+		go func(url string) {
+			defer wg.Done()
+			id, err := getImageUrlForArtist(url)
+			if err != nil {
+				log.Println("Error getting image url for artist", artist.Name, err)
+				return
 			}
-		}
-
+			newArtist.ImageUrl = "https://lastfm.freetls.fastly.net/i/u/300x300/" + id
```

Comment:
Maybe leave a comment about what's happening here

Reply:
Not necessarily here, but the somewhere explaining the reasoning behind scraping and getting the ID